documentation/src/main/asciidoc/integrationguide/chapters/services/extras/override/ex3-contributor.java

```diff
@@ -3,17 +3,17 @@ public class LatestAndGreatestConnectionProviderImplContributor
 
     @Override
     public void contribute(
-            standardserviceregistrybuilder serviceregistrybuilder) {
+            StandardServiceRegistryBuilder serviceRegistryBuilder) {
 
         // here we will register a short-name for our service strategy
-        strategyselector selector = serviceregistrybuilder
-            .getbootstrapserviceregistry().
-            .getservice( strategyselector.class );
+        StrategySelector selector = serviceRegistryBuilder
+            .getBootstrapServiceRegistry().
+            .getService( StrategySelector.class );
 
-        selector.registerstrategyimplementor(
-            connectionprovider.class,
-            "lag"
-            latestandgreatestconnectionproviderimpl.class
+        selector.registerStrategyImplementor(
+            ConnectionProvider.class,
+            "lag",
+            LatestAndGreatestConnectionProviderImpl.class
         );
     }
```

Comment:
this code snippet is too sloppy

Reply:
Probably should be `requireService()` instead of `getService()` if we're being picky.

Reply:
yeah, we should use that, otherwise NPE later can't be avoided.